### PR TITLE
Addresses cargo-pgrx error reporting

### PR DESCRIPTION
**Pull request by BradyBonnette**
There are some situations where `cargo pgrx` reports an error but important details are missing, as it relates to the spawning of child process. One example of this is when `cargo pgrx` downloads and compiles PostgreSQL from source, as demonstrated by https://github.com/pgcentralfoundation/pgrx/issues/1222

The changes here seek to make any error output due to a spawned child process be reported to the end user instead of being silently swallowed and disregarded.

### CODE REVIEW

**Review by thomcc — APPROVED** [review]
LGTM

**Comment by BradyBonnette** [comment]
@thomcc I might go through some of the other places where `Command` is used and see if I can verify that those report similar to this.

File: cargo-pgrx/src/command/init.rs [hunk 1]
```diff
@@ -271,14 +271,35 @@ fn untar(bytes: &[u8], pgrxdir: &PathBuf, pg_config: &PgConfig) -> eyre::Result<
...
+    // `.flush()` call immediately after.
+    match stdin.write_all(bytes) {
+        Ok(_) => (),
+        Err(e) => error = Some(e),
+    }
+
+    match stdin.flush() {
+        Ok(_) => (),
+        Err(e) => error = Some(e),
+    }
```
Comment (workingjubilee) [inline cargo-pgrx/src/command/init.rs:288]: Is it possible for both of these to error? We'd hypothetically be dumping one error on the floor in favor of another, again.

**Review by workingjubilee — COMMENTED** [review]
Yeah it _might_ be nice to factorize this out a bit if we have a bunch of `Command::spawn` sites that look like this. Not strictly necessary, tho'.

File: cargo-pgrx/src/command/init.rs (hunk 1 above)
Comment (thomcc) [reply cargo-pgrx/src/command/init.rs:288]: I would be very surprised if flush did anything on this type. But we should probably either take the first error or use chaining, if we care.

Comment (BradyBonnette) [reply cargo-pgrx/src/command/init.rs:288]: I thought about this as I was writing it, but I wasn't entirely sure how to handle it without making it "shift right" or making a plate of spaghetti out of it.

Comment (workingjubilee) [reply cargo-pgrx/src/command/init.rs:288]: You will see me tend to incrementally construct and then immediately pattern-match on tuples a lot in my code for this reason.

**Pull request merged**